src/DO/Sara/ImageProcessing/Deriche.hpp

```diff
@@ -228,24 +227,18 @@ namespace DO { namespace Sara {
     return J;
   }
 
-  //! \brief Helper class to use: Image<T,N>::compute<DericheBlur>(T sigma)
-  template <typename T, int N>
+  //! @brief Wrapper class to use: Image<T,N>::compute<DericheBlur>(T sigma)
   struct DericheBlur
   {
-    typedef Image<T, N> return_type;
-    typedef typename PixelTraits<T>::channel_type parameter_type;
+    template <typename Image>
+    using ReturnType = Image;
 
-    inline explicit DericheBlur(const Image<T, N>& src)
-      : src_(src)
+    template <typename Image, typename Sigma>
```

Comment:
The operator overloading would allow a more versatile reuse here since it is templated.